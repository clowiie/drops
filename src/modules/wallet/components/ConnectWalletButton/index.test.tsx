import { render } from '@testing-library/react'

import createRouteWrapper from 'test/utils/createRouterWrapper'

import * as ConnectWalletButtonType from '.'

describe('ConnectWalletButton', () => {
  const { default: ConnectWalletButton } =
    require('.') as typeof ConnectWalletButtonType

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container } = render(<ConnectWalletButton />, {
        wrapper: createRouteWrapper(
          {
            initialIndex: 0,
            initialEntries: ['/somepath'],
          },
          {},
        ),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(container.querySelector('a')?.href).toBe(
        'http://localhost/somepath?modalConnectWallet=true',
      )
    })
  })
})
