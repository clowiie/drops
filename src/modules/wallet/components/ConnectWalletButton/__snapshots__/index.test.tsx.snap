// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectWalletButton snapshots should match snapshots 1`] = `
<a
  class="bg-gray-700 hover:bg-gray-500 text-white p-2 rounded-lg text-sm"
  href="/somepath?modalConnectWallet=true"
>
  Connect Wallet
</a>
`;
